src/lib.rs

```diff
@@ -622,6 +668,62 @@ where
     }
 }
 
+impl<T, N> Counter<T, N>
+where
+    T: Hash + Eq,
+    N: PartialOrd + Zero,
+{
+    /// Test wether this counter is a superset of another counter.
+    /// This is true if for all elements in this counter and the other,
+    /// the count in this counter is greater than or equal to the count in the other.
+    ///
+    /// `c.is_superset(&d);` -> `c.iter().all(|(x, n)| n >= d[x]) && d.iter().all(|(x, n)| c[x] >= n)`
+    ///
+    /// ```rust
+    /// # use counter::Counter;
+    /// # use std::collections::HashMap;
+    /// let c = "aaabbc".chars().collect::<Counter<_>>();
+    /// let mut d = "abb".chars().collect::<Counter<_>>();
+    ///
+    /// assert!(c.is_superset(&d));
+    /// d[&'e'] = 1;
+    /// assert!(!c.is_superset(&d));
+    /// ```
+    pub fn is_superset(&self, other: &Self) -> bool {
+        // need to test keys from both counters, because if N is signed, counts in `self`
+        // could be < 0 for elements missing in `other`. For the unsigned case, only elements
+        // from `other` would need to be tested.
+        self.keys()
+            .chain(other.keys())
+            .all(|key| self[key] >= other[key])
```

Comment:
Should use `.get(key)` here instead of indexing, because a failed indexing operation panics.

Reply:
> because a failed indexing operation panics.

Really? I'll have to take your word for it!

The way I traced it (reading the code and in a debugger) the Index trait implementation for Counter uses `self.map.get(k).unwrap_or(&self.zero)`; is there any way the trait wouldn't be used here? Only I'd otherwise would have used `self.map.get(k).or(Some(&self.zero)) >= other.map.get(k).or(Some(&other.zero))` here which isn't much of a difference.

Reply:
Oh wow, I'd forgotten about that implementation detail. I was thinking that it was accessing the underlying [`HashMap` implementation](https://doc.rust-lang.org/stable/std/collections/struct.HashMap.html#panics-1) via `Deref`, which does panic. Nevermind, I was wrong.